portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,25 +27,39 @@
 /** Servlet that returns an Array of comments as JSON. TODO: modify this file to handle comments data from user input */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    ArrayList<String> comments = new ArrayList<>();
 
   @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String message;
-         
-        ArrayList<String> greetings = new ArrayList<>();
-        greetings.add("Hey cutie, I like your hoodie. Dont lose the goodies");
-        greetings.add("Hey cutie, I like your hat. Careful with the bats");
-        greetings.add("Hey cutie, I like your anti-capitalist shirt, lets go take down the bourgeoisie");
-        
-        String json = convertToJson(greetings);
-
+        String json = convertToJson(comments);
         response.setContentType("application/json;");
         response.getWriter().println(json);
     }
 
-    private String convertToJson(ArrayList greetings) {
+    private String convertToJson(ArrayList comment) {
         Gson gson = new Gson();
-        String json = gson.toJson(greetings);
+        String json = gson.toJson(comment);
         return json;
     }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String text = getParameter(request, "text-input", "");
+        comments.add(text);
+
+        String json = convertToJson(comments);
+    
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
+
+        response.sendRedirect("/index.html");
+    }
+
+    private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
You can make this easier to read with `Optional`

```java
private String getParameter(HttpServletRequest request, String name, String defaultValue) {
  return Optional.ofNullable(request.getParameter(name)).orElse(defaultValue);
}
```

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html

Reply:
This line of code broke my code. When I try to run the environment it gives me the error: cannot find symbol
[ERROR]   symbol:   variable Optional
[ERROR]   location: class com.google.sps.servlets.DataServlet

I assume my problem lies in how I define the global variable in order to use Optional. 
Public Static <T> Optional<T> ofNullable(T value); ?

Reply:
No you don't need to define any variable, just import the package, it's built into java

```java
import java.util.Optional

// some classes and things

Optional.ofNullable(request.getParameter(name)).orElse(defaultValue);
```